backend.py

```diff
@@ -0,0 +1,30 @@
+# Class/Backend
+
+class Backend:
+    def construct_query(self, search_parameters: list[tuple]) -> str:
+        pass
+            # Constructs a query string from the given search parameters
+            # Arguments: search_parameters (list[tuple]): A list of tuples representing the searc h parameters
+            
+            # Returns: str: The constructed query string
+
+
+    def query_api(self, query: str)->list['Card']:
+        pass
+            # Queries the API using the constructed query string
+            #Arguments: query(str): The query string to send to the API
+
+            # Returns: list[Card]: A list of Card objects returned by the API
+
+
+
+    def all_sets(self) -> list['Set']:
+        
+            # Retrieves all available sets from the API
+
+            # Returns: list[Sets]; A list of Set objects retrieved from the API
+
+
+        sets_list = Set.all()
```

Comment:
- [ ] **Implementation**: Shouldn't push stuff beyond the work task. And try to never commit things outside of the scope of the work task. You can submit issues for things like typos, or other things that you notice needing fixed. Try to stick with only the scope of what you are tasked to do per commit/push.

We'll leave it here this time. Just clean up the stuff I mentioned above in each function, and clean up the extra white spaces.

Reply:
Actually it's better to remove this for now. We don't want to do more than the step we're on, this is for many reasons including Organization, Commit History Clarity, etc.